### DISCUSSION TITLE
remove_dynablocklist

### DISCUSSION THREAD

**Discussion by 1804020220** [ptitSeb/box86#808] [Q&A]
Why does this branch (**remove_dynablocklist**) take up less memory than the main branch and doesn't run slower?
Does anyone know exactly how it works?

**Comment by ptitSeb** [comment]
That was an experiment from some time ago. I should have used some of the learnings from this branch in main. But I can recheck if I forgot something.
I didn't merged the branch because I had stability issues with it (I think Unity3D/mono games were not running fine, but I don't remember exactly).

**Reply by 1804020220** [reply]
I met with this [issue ](https://github.com/ptitSeb/box86/issues/745)a similar problem, but I used this branch remove_dynablocklist compiler runs, found the runtime memory less occupied, The running speed is normal. Does this mean that the dynablocklist consumes a lot of memory at runtime?

**Reply by ptitSeb** [reply]
dynablocklist maps memory used be x86 code, so it can take some memory for large program yes. That's why I did some experiment to remove them.
On box64 for example, I completly removed them, but I also needed to removed the "child" concept, so a bloxk of converted code can only be joined on the start address, no is some other point.
I'm usure if it this should be done for box86 tbh. It's quite some work, and it was really needed on box64 as the memory consuption was too high. Maybe I need to work on that on box86 too?

**Reply by 1804020220** [reply]
You may also need to remove the dynablocklist from box86, which takes up a lot of memory when running a lot of programs, and after a while the system freezes. Maybe after solving the Unity3D game problem you mentioned above, we can combine this branch into the master branch?
